### DISCUSSION TITLE
2 timers

### DISCUSSION THREAD

**Discussion by DesmoricFR** [ncosgray/cuppa_mobile#63] [Ideas]
It will be usefull to use 2 timers in a same time in case of we have to prepare 2 different teas. 
Do you think it’s possible ?

Regards

**Comment by ncosgray** [comment]
Wow, this is a really excellent idea! It should be possible. I've realized it would not be super complicated, particularly if I can come up with a good way to update the interface to accommodate 2 timers.

I'm thinking for those who are only timing 1 tea the interface would stay exactly the same as it is now. But if there is already a timer running and the user taps a tea button, Cuppa could ask if they want to cancel (current behavior) or start a second timer (new behavior). In the latter case, Cuppa's interface would then adapt. Instead of one countdown timer at the top of the screen, the green box would expand to include a second countdown timer below the first. And a second teacup with a teabag would appear alongside the first cup in the center of the screen. I am not sure yet how the Cancel button would adapt.

What are your thoughts? This might look similar to the concept below:

|1 timer|2 timers|
|-------|--------|
|![Cuppa-1timer](https://user-images.githubusercontent.com/13181064/150697217-d73380cb-9292-470d-8f53-1b60f14f68c2.png)|![Cuppa-2timer-concept](https://user-images.githubusercontent.com/13181064/150697829-65e7e0cf-69cc-4a18-b4ed-d5f816ea591d.png)|

**Reply by DesmoricFR** [reply]
It's great ! Nice design for the timer and the 2 cups. I've haven't any precise idea on my mind and you did a very good work. I think you're on the right way :)

**Comment by ncosgray** [comment]
I'm happy to announce that dual timers are finally here with the latest version of Cuppa!

https://github.com/ncosgray/cuppa_mobile/releases/tag/1.8.0

**Reply by DesmoricFR** [reply]
Hello,
Very nice work for this new version with 2 timers. Really simple to use, just wonderful.
Congratulation 😃
